api.c

```diff
@@ -521,6 +521,14 @@ int tcmu_emulate_evpd_inquiry(
 		/* Optimal xfer length */
 		memcpy(&data[12], &val32, 4);
 
+		/* MAXIMUM UNMAP LBA COUNT */
+		val32 = htobe32(VPD_MAX_UNMAP_LBA_COUNT);
+		memcpy(&data[20], &val32, 4);
+
+		/* MAXIMUM UNMAP BLOCK DESCRIPTOR COUNT */
+		val32 = htobe32(VPD_MAX_UNMAP_BLOCK_DESC_COUNT);
+		memcpy(&data[24], &val32, 4);
+
 		tcmu_memcpy_into_iovec(iovec, iov_cnt, data, sizeof(data));
 
 		return SAM_STAT_GOOD;
```

Comment:
You also need to set the page length to 0x3c when setting the thinp related values. If thinp is not used them it is just 0x10.

Linux' checks for this in sd_read_block_limits().